lib/Zonemaster/Backend/DB.pm

```diff
@@ -537,6 +537,7 @@ sub _project_params {
     my %projection = ();
 
     $projection{domain}   = lc( $$params{domain}  // "" );
+    $projection{domain}   =~ s/\.$// unless $projection{domain} eq '.';
```

Comment:
`test.xa..` and `..` are invalid domains, they should not be turned into valid domains arbitrarily, they should instead be rejected by the validation step (which is not done currently).

Reply:
I agree. But then this code will turn ".." and "test.xa.." into "." and "test.xa." respectively.

Until we have code that rejects domain names with consecutive dots with a meaningful message, I think it is reasonable to fix to avoid crashes.

Reply:
I think I do agree with @blacksponge here. One final dot is a valid input (that we need to clean to process it internally). However more consecutive dots is erroneous. There is no need to clean the input in this case.

Reply:
I also agree with both of you that double dot is an error, which should be captured and resulted in an error message. But that is not what will happen. Zonemaster will crash, and I prefer that Zonemaster for a while does more than it should so it does not crash.

See it as a fix while waiting for the correct solution.

Reply:
If all we need is to check for empty labels i can add
```perl
if ( $domain =~ m/\.\./i ) {
    return  N__ 'The domain contains empty labels';
}
```
to `check_domain`, I think that would be better.

Reply:
this would produce errors like
```
./script/zmb start_domain_test --domain zonemaster..net.  --nameserver ns2.nic.fr...: | jq                                                                              ✔  16.13.0   gbm@D0023
{
  "error": {
    "data": [
      {
        "path": "/domain",
        "message": "The domain contains empty labels"
      },
      {
        "path": "/nameservers/0/ip",
        "message": "Invalid IP address"
      },
      {
        "message": "The domain contains empty labels",
        "path": "/nameservers/0/ns"
      }
    ],
    "code": "-32602",
    "message": "Invalid method parameter(s)."
  },
  "jsonrpc": "2.0",
  "id": 1
}
```

Reply:
I like the idea, it is more of a solution than fix. The error message is technically (almost*) correct, but probably hard to understand. I suggest:
```perl
    return  N__ 'The domain is invalid: it contains consecutive dots';
```

(*) All domain names ends with an empty label after the final dot, representing the root.